src/hooks/useInterval.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useRef } from "react";
+
+/**
+ * React hook that simplifies using setInterval() within functional components.
+ *
+ * Note: You don't need to memoise/useCallback the handler as we store it with
+ * a ref internally.
+ *
+ * @param handler the function to be called every tick of the interval
+ * @param intervalInMs the interval time in milliseconds
+ */
+export const useInterval = (handler: () => void, intervalInMs?: number) => {
```

Comment:
```suggestion
/**
 * React hook that simplifies using setInterval() within functional components.
 * If `null` or `undefined` are supplied `intervalInMs`, the hook will do nothing.
 * You can use this to avoid conditionally calling the hook.
 *
 * Note: You don't need to memoise/useCallback the handler as we store it with
 * a ref internally.
 *
 * @param handler the function to be called every tick of the interval
 * @param intervalInMs the interval time in milliseconds. If falsy, the hook
 * will do nothing.
 */
export const useInterval = (handler: () => void, intervalInMs?: number) => {
```